src/it/compare-pom/verify.groovy

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+class LogInspector
+{
+    File log;
+    int index;
+    LogInspector( File log )
+    {
+        this.log = log;
+        this.index = 0;
+    }
+    boolean containsAfter( CharSequence text )
+    {
+        int newIdx = log.text.indexOf( text, index + 1 )
+        if ( newIdx > index)
+        {
+            index = newIdx
+            return true
+        }
+        return false
+    }
+    String toString ()
+    {
+        return "Log file ${log} after index ${index}."
+    }
+}
+
+
+assert new File( basedir, "repo/org/apache/maven/its/deploy/comparepom/test/maven-metadata.xml" ).exists()
+
+File snDir = new File( basedir, "repo/org/apache/maven/its/deploy/comparepom/test/1.0-SNAPSHOT/" )
+assert snDir.list( {d, f -> f ==~ /test-1.0-.*-first.jar/} as FilenameFilter ).size() == 1
+assert snDir.list( {d, f -> f ==~ /test-1.0-.*-second.jar/} as FilenameFilter ).size() == 1
+assert snDir.list( {d, f -> f ==~ /test-1.0-.*.pom/} as FilenameFilter ).size() == 2
+
+assert new File( basedir, "repo/org/apache/maven/its/deploy/comparepom/test/1.0/test-1.0-first.jar" ).exists()
+assert new File( basedir, "repo/org/apache/maven/its/deploy/comparepom/test/1.0/test-1.0-second.jar").exists()
+
+File deployedPom = new File( basedir, "repo/org/apache/maven/its/deploy/comparepom/test/1.0/test-1.0.pom" )
+assert deployedPom.exists()
+assert ! deployedPom.text.contains("Modified POM!")
+
+File installedPom = new File( localRepositoryPath, "org/apache/maven/its/deploy/comparepom/test/1.0/test-1.0.pom" )
+assert installedPom.exists()
+assert installedPom.text.contains("Modified POM!")
+
+File buildLog = new File( basedir, 'build.log' )
+assert buildLog.exists()
+
+// Inspect log
+LogInspector li = new LogInspector( buildLog )
+String groupUrl = "file:///${basedir}/repo/org/apache/maven/its/deploy/comparepom"
+
+// 1st and 2nd run: The POM tried to be downloaded and uploaded:
```

Comment:
Log messages are not reliable over time. Tests should not use them to inspect behavior.